prov/rxm/src/rxm.h

```diff
@@ -307,7 +320,14 @@ struct rxm_recv_entry {
 	uint64_t ignore;
 	uint64_t comp_flags;
 	size_t total_len;
+	struct rxm_recv_queue *recv_queue;
 	void *multi_recv_buf;
+
+	/* Used for sequetila receive of data */
+	size_t total_recv_len;
+	uint64_t msg_id;
+	RbtIterator rbt_iter;
+	size_t last_recv_seg_num;
```

Comment:
See comment for rxm_tx_entry.  Nit: spelling of sequential.

I guess I'll figure this out reading down, but is a rbtree for this really necessary?  I wouldn't expect to receive out of order data or any missing data.  A simple sorted list may be all that's needed, with a strong likelihood (maybe even guaranteed) that all new messages will be added to the tail.

Reply:
It seems the rbtree is being used to retrieve the correct recv entry so that tag matching and releated processing is done only once per msg_id.